### PR TITLE
Added solution

### PR DESCRIPTION
**Pull request by AndrewPolishchuc**

### CODE REVIEW

File: src/main/java/movie/storage/service/impl/OrderServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    @Inject
+    private ShoppingCartService shoppingCartService;
+
+    @Override
+    public Order completeOrder(List<Ticket> tickets, User user) {
+        Order order = new Order();
+        order.setUser(user);
+        order.setTickets(tickets);
+        shoppingCartService.clear(shoppingCartService.getByUser(user));
+        return orderDao.add(order);
```
Comment (andromash) [inline src/main/java/movie/storage/service/impl/OrderServiceImpl.java:29]: Why did you choose not to set the date? Or is it created automatically?

**Review by andromash — COMMENTED** [review]
Nice job!

**Review by yutsevich — APPROVED** [review]
👍

File: src/main/java/movie/storage/dao/impl/OrderDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            }
+        }
+    }
+
+    @Override
+    public List<Order> getOrderHistory(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("from Order o "
+                    + "left join fetch o.tickets "
+                    + "join fetch o.user "
```
Comment (dmytro-hryhoruk) [inline src/main/java/movie/storage/dao/impl/OrderDaoImpl.java]: If you order has more than one ticket, will you have any duplications in your list?

File: src/main/java/movie/storage/model/Order.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+@Entity
+@Table(name = "orders")
+public class Order {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @OneToMany
+    private List<Ticket> tickets;
+    @ManyToOne(fetch = FetchType.LAZY)
+    private User user;
```
Comment (dmytro-hryhoruk) [inline src/main/java/movie/storage/model/Order.java]: what's about orderDate?

File: src/main/java/movie/storage/service/impl/OrderServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    @Inject
+    private OrderDao orderDao;
+    @Inject
+    private ShoppingCartService shoppingCartService;
+
+    @Override
+    public Order completeOrder(List<Ticket> tickets, User user) {
+        Order order = new Order();
+        order.setUser(user);
+        order.setTickets(tickets);
```
Comment (dmytro-hryhoruk) [inline src/main/java/movie/storage/service/impl/OrderServiceImpl.java]: orderDao.add() will also create a delete statement for shoppingCart tickets(as in shoppingCartService.clear). Let's use deep cloning here

File: src/main/java/movie/storage/dao/impl/OrderDaoImpl.java (hunk 2 above)
Comment (AndrewPolishchuc) [reply src/main/java/movie/storage/dao/impl/OrderDaoImpl.java]: Done

File: src/main/java/movie/storage/model/Order.java (hunk 3 above)
Comment (AndrewPolishchuc) [reply src/main/java/movie/storage/model/Order.java]: Added

File: src/main/java/movie/storage/service/impl/OrderServiceImpl.java (hunk 4 above)
Comment (AndrewPolishchuc) [reply src/main/java/movie/storage/service/impl/OrderServiceImpl.java]: Done

File: src/main/java/movie/storage/service/impl/OrderServiceImpl.java (hunk 1 above)
Comment (AndrewPolishchuc) [reply src/main/java/movie/storage/service/impl/OrderServiceImpl.java:29]: Added

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**